core/referral/engine.go

```diff
@@ -558,6 +558,10 @@ func (e *Engine) computeFactorsByReferee(ctx context.Context, epoch uint64, take
 		e.factorsByReferee[party] = refereeStats
 		setStats.RefereesStats[party] = refereeStats
 
+		if e.isSetEligible(setID) != nil {
```

Comment:
why go all the way to here rather than do this in line 537 for example?

Reply:
Because I wanted these:
```
 558        e.factorsByReferee[party] = refereeStats
 559        setStats.RefereesStats[party] = refereeStats
```
To be set with set.

Reply:
but why? they'll just be 0 won't they?

Reply:
yes, but we will want the taker volume stats to be stored accross the epoch and sent to the datanode.

Reply:
basically we still want the refereeStats to be updated.